src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideLogsStep.groovy

```diff
@@ -0,0 +1,230 @@
+         * Logs a warning if no reports were returned by ecu.test.
+         * @return List of strings containing the paths of the logs.
+         */
+        @Override
+        Void call() throws IOException {
+            listener.logger.println("Providing ecu.test logs to jenkins.")
+            try {
+                RestApiClient apiClient = RestApiClientFactory.getRestApiClient(envVars.get('ET_API_HOSTNAME'), envVars.get('ET_API_PORT'))
+                List<ReportInfo> reports = apiClient.getAllReports()
+
+                if (reports == null || reports.isEmpty()) {
+                    listener.logger.println("[WARNING] No report files returned by ecu.test")
+                }
+                for (def report : reports) {
+                    String reportDir = report.reportDir.split('/').last()
+                    if (!checkReportFolderCreationDate(reportDir)) {
+                        listener.logger.println("[WARNING] ecu.test reports contains folder older than this run. Path: ${report.reportDir}")
+                    }
+
+                    File reportFolderZip = apiClient.downloadReportFolder(report.testReportId)
+                    if (!extractAndSaveFromZip(reportFolderZip, ["ecu.test_out", "ecu.test_err"], "${logDirPath}/${reportDir}")) {
+                        throw new AbortException("${report.reportDir} is missing one or all log files!")
+                    }
+                }
+            }
+            catch (ApiException e) {
+                if (e instanceof de.tracetronic.cxs.generated.et.client.v2.ApiException && e.code == 404) {
+                    throw new AbortException("[ERROR] Downloading reportFolder is not available for ecu.test < 2024.2!")
+                } else {
+                    throw new AbortException(e.message)
+                }
+            }
+            listener.logger.flush()
+        }
+
+
+        /**
+         * Checks if given reportDir name, which includes the date was created after this build was started
+         * @return boolean
+         */
+        boolean checkReportFolderCreationDate(String reportDir) {
+            String df = "yyyy-MM-dd_HHmmss"
+            SimpleDateFormat dateFormat = new SimpleDateFormat(df)
+            String pattern = /\d{4}-\d{2}-\d{2}_\d{6}/
+            def matcher = reportDir =~ pattern
+            if (matcher.find()) {
+                String matchedText = matcher.group(0)
+                return dateFormat.parse(matchedText) > new Date(startTimeMillis)
+            }
+            listener.logger.println("[WARNING] Could not extract date from report folder name: ${reportDir}")
+            return false
+        }
+
+        /**
+         * Extracts and saves files containing the given strings in their name out of a given zip folder to given path.
+         * @return boolean
+         */
+        static boolean extractAndSaveFromZip(File reportFolderZip, List<String> extractFilesContaining, String saveToPath) {
```

Comment:
use native functions like [unzip](https://www.jenkins.io/doc/pipeline/steps/pipeline-utility-steps/#unzip-extract-zip-file)

Reply:
wouldn't we have to add org.jenkinsci.plugins.pipeline.utility.steps to our dependencies to use this ?